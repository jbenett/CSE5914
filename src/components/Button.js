import styled from "styled-components";

const Button = styled.div`
    display: inline-block;
    padding: 0.5rem 0.75rem;
    background-color: white;
    border-radius: 3px;
    font-family: ${props => props.theme.fontFamily};
    font-weight: bold;
    transition: 150ms all;
    ${props => props.theme.dropShadow.subtle}

    &:hover {
        cursor: pointer;
        color: white;
        background-color: ${props => props.theme.primary};
        ${props => props.theme.dropShadow.regular}
    }
`;

export default Button;