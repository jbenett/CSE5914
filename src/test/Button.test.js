import React from 'react';
import { shallow } from 'enzyme';
import Button from '../components/Button.js'

describe('Button', () => {
  it('should render correctly', () => {
    const component = shallow(<Button />);
    expect(component).toMatchSnapshot();
  });
});
