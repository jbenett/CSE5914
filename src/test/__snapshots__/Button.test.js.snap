// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "forwardedComponent": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
    display: inline-block;
    padding: 0.5rem 0.75rem;
    background-color: white;
    border-radius: 3px;
    font-family: ",
            [Function],
            ";
    font-weight: bold;
    transition: 150ms all;
    ",
            [Function],
            "

    &:hover {
        cursor: pointer;
        color: white;
        background-color: ",
            [Function],
            ";
        ",
            [Function],
            "
    }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "forwardedRef": null,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "forwardedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "rules": Array [
              "
    display: inline-block;
    padding: 0.5rem 0.75rem;
    background-color: white;
    border-radius: 3px;
    font-family: ",
              [Function],
              ";
    font-weight: bold;
    transition: 150ms all;
    ",
              [Function],
              "

    &:hover {
        cursor: pointer;
        color: white;
        background-color: ",
              [Function],
              ";
        ",
              [Function],
              "
    }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
